polygeist/tools/cgeist/driver.cc

```diff
@@ -395,6 +395,15 @@ static LogicalResult canonicalize(mlir::MLIRContext &Ctx,
   return success();
 }
 
+static bool shouldEarlyDropHostCode() {
+  // We can drop host code early if both conditions apply:
+  // 1. -no-early-drop-host-code is not passed;
+  // 2. Host code isn't explicitly asked for when we emit MLIR.
+  return !NoEarlyDropHostCode && (SYCLDeviceOnly ||
+                                  // We do not emit MLIR
+                                  EmitLLVM || EmitOpenMPIR || EmitAssembly);
```

Comment:
On https://github.com/intel/llvm/blob/b4bf664e1e7ddf46428abd913be52975a3e78d50/polygeist/tools/cgeist/driver.cc#L928, `bool EmitMLIR = EmitAssembly && !EmitLLVM;`, so `EmitAssembly` is true when we emit MLIR.